lib/deserializer-utils.js

```diff
@@ -42,7 +42,7 @@ module.exports = function (jsonapi, data, opts) {
         relationshipName,
         relationshipData.type,
         relationshipData.id,
-      ]
+      ].join('_');
```

Comment:
This wouldn't work, a path should use `].join('.');`

Reply:
ok I see what's happening here. This would be a more complete refactor:
```
      var path = [
        from.type,
        from.id,
        relationshipName,
        relationshipData.type,
        relationshipData.id,
      ].join('_');

      // Check if the include is already processed (prevent circular references).
      if (alreadyIncluded[path]) return resolve(null);
      alreadyIncluded[path] = true;
```
plus the removal of the lodash get, set & merge requires.